src/dswx_sar/metadata.py

```diff
@@ -0,0 +1,268 @@
+import os
+import h5py
+import glob
+from datetime import datetime
+from collections import OrderedDict
+from dswx_sar.version import VERSION as SOFTWARE_VERSION
+
+def _copy_meta_data_from_rtc(h5path_list, dswx_metadata_dict):
+    """Copy metadata dictionary from RTC metadata
+    Parameters
+    ----------
+    h5path_list : list
+            List of metadata HDF5 file
+    dswx_metadata_dict : collections.OrderedDict
+        Metadata dictionary
+    """
+
+    metadata_dict = dict()
+
+    metadata_rtc_str_list = [
+        'orbitPassDirection',
+        'burstID',
+        'productVersion', 
+        'zeroDopplerStartTime']
+    metadata_rtc_int_list = [
+        'trackNumber',
+        'absoluteOrbitNumber']
+
+    metadata_rtc_list = metadata_rtc_str_list + metadata_rtc_int_list
+    dswx_metadata_list = ['ORBIT_PASS_DIRECTION',
+                          'BURST_ID',
+                          'RTC_PRODUCT_VERSION',
+                          'SENSING_TIME',
+                          'TRACK_NUMBER',
+                          'ABSOLUTE_ORBIT_NUMBER']
+
+    for metadata in metadata_rtc_list:
+        metadata_dict[metadata] = []
+
+    for h5_meta_path in h5path_list:
+        with h5py.File(h5_meta_path) as src:
+
+            spacecraft_name = list(src['/science/'].keys())[0]
+            identification_path = f'/science/{spacecraft_name}/identification/'
+
+            for metadata_field in metadata_rtc_str_list:
+                metadata_dict[metadata_field].append(
+                    str(src[f'{identification_path}/{metadata_field}'].asstr()[...]))
+
+            for metadata_field in metadata_rtc_int_list:
+                metadata_dict[metadata_field].append(
+                    str(src[f'{identification_path}/{metadata_field}'][...]))
+
+    for metadata, dswx_meta_field in zip(metadata_rtc_list, dswx_metadata_list):
+
+        if dswx_meta_field == 'SENSING_TIME':
+            input_date_format = "%Y-%m-%dT%H:%M:%S"
+            output_date_format = "%Y-%m-%dT%H:%M:%SZ"
+
+            date_list = [datetime.strptime(date_single[0: 19], input_date_format) 
+                                for date_single in metadata_dict[metadata]]
+            dswx_metadata_dict['SENSING_START'] = datetime.strftime(min(date_list), 
+                                                    output_date_format)
+            dswx_metadata_dict['SENSING_END'] = datetime.strftime(max(date_list), 
+                                                    output_date_format)
```

Comment:
My intention here for `sensing_start` and sensing `end` is that all MGRS tiles within the collections of MGRS tile share the same `sensing_start` and `sensing_end`. This is because they are simultaneously used to compute the threshold. Of course, it is possible to assign the different sensing time for the individual MGRS grids by finding the intersecting bursts.